include/glow/Graph/Graph.h

```diff
@@ -196,6 +196,13 @@ class Module final {
 
   ///@}
 
+  /// Creates a new constant in the graph with the given \p name by slicing an
+  /// existing \p constant using the given \p offset and \p dims. \returns the
+  /// newly created constant.
+  Constant *createConstantSlice(llvm::StringRef name, Constant *constant,
```

Comment:
The reason I created Constant slices instead of Slice nodes is because the graph optimizer was not always doing the extracts at compile time (most of the slice extractions are also followed by transposes so it is critical to force them to occur at compile time). Now the truth is that this was the exhibited behavior some time ago (this feature was implemented some months ago) so I don`t know this is the case currently.

Reply:
I verified that with the new optimization infrastructure this "manual" slicing of the constant is no longer necessary.  Therefore I removed this function along with it's invocations and used plain "createSlice" invocations in the code. I also promoted the LSTM input variable types from Constant (which was the case until now) to generic NodeValue. Thanks for the proposal!